src/modules/Modal/Modal.js

```diff
@@ -133,8 +148,13 @@ class Modal extends Component {
       }
     }
 
-    document.addEventListener('keydown', this.handleDocumentKeyDown)
-    document.addEventListener('click', this.handleClickOutside)
+    if (closeOnEscape) {
+      document.addEventListener('keydown', this.handleDocumentKeyDown)
+    }
```

Comment:
You have a valid point. I will change to add listener always but take action conditionally.